test/gtest/conv_igemm_dynamic_xdlops_nhwc_bf16.cpp

```diff
@@ -0,0 +1,220 @@
+    while(begin != end)
+        tokens.push_back(*begin++);
+}
+
+class Conv2dBf16 : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+void Run2dDriver(miopenDataType_t prec)
+{
+
+    std::vector<TestCase> params;
+    switch(prec)
+    {
+    case miopenBFloat16: params = Conv2dBf16::GetParam(); break;
+    case miopenFloat:
+    case miopenHalf:
+    case miopenInt8:
+    case miopenInt8x4:
+    case miopenInt32:
+    case miopenDouble:
+    case miopenFloat8:
+    case miopenBFloat8:
+        FAIL() << "miopenFloat, miopenHalf, miopenInt8, miopenInt8x4, miopenInt32, "
+                  "miopenDouble, miopenFloat8, miopenBFloat8 "
+                  "data type not supported by conv_igemm_dynamic_xdlops_nhwc_bf16 test";
+
+    default: params = Conv2dBf16::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value, tokens);
+        std::vector<const char*> ptrs;
+
+        std::transform(tokens.begin(),
+                       tokens.end(),
+                       std::back_inserter(ptrs),
+                       [](const std::string& str) { return str.data(); });
+
+        testing::internal::CaptureStderr();
+        test_drive<conv2d_driver>(ptrs.size(), ptrs.data());
+        auto capture = testing::internal::GetCapturedStderr();
+        std::cout << capture;
+    }
+};
+
+bool IsTestSupportedForDevice(const miopen::Handle& handle)
+{
+    std::string devName = handle.GetDeviceName();
+    if(devName == "gfx90a" || miopen::StartsWith(devName, "gfx94"))
+        return true;
+    else
+        return false;
+}
+
+TEST_P(Conv2dBf16, Bf16Test)
+{
+    const auto& handle = get_handle();
+    if(IsTestSupportedForDevice(handle) && !SkipTest())
+    {
+        Run2dDriver(miopenBFloat16);
+    }
+    else
+    {
+        GTEST_SKIP();
+    }
+};
+
+std::vector<TestCase> GetTestCases(const std::string& precision)
+{
+    std::vector<std::string> env_nhwc_fwd = {
+        "MIOPEN_FIND_MODE=normal",
+        "MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvAsmImplicitGemmGTCDynamicFwdXdlopsNHWC"};
+
+    std::vector<std::string> env_nhwc_bwd = {
+        "MIOPEN_FIND_MODE=normal",
+        "MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvAsmImplicitGemmGTCDynamicBwdXdlopsNHWC"};
+
+    std::vector<std::string> env_nhwc_wrw = {
+        "MIOPEN_FIND_MODE=normal",
+        "MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvAsmImplicitGemmGTCDynamicWrwXdlopsNHWC"};
+
+    std::string v             = " --verbose";
+    std::string dis_bk_data   = " --disable-backward-data";
+    std::string dis_bk_wei    = " --disable-backward-weights";
+    std::string dis_fwd       = " --disable-forward";
+    std::string dis_vali      = " --disable-validation";
+    std::string in_nhwc       = " --in_layout NHWC";
+    std::string fil_nhwc      = " --fil_layout NHWC";
+    std::string out_nhwc      = " --out_layout NHWC";
+    std::string args_nhwc_wrw = dis_fwd + dis_bk_data + in_nhwc + fil_nhwc + out_nhwc;
```

Comment:
It looks like the other directions and the other tests may get benefits if all the common option would be combined like this wrw case.
For example:
```cpp
const std::string args_nhwc_fwd = dis_bk_data + dis_bk_wei + in_nhwc + fil_nhwc + out_nhwc;
const std::string args_nhwc_bwd = dis_fwd + dis_bk_wei + in_nhwc + fil_nhwc + out_nhwc;
```